### ISSUE TITLE
Loading spinner not shown on stalls

### ISSUE THREAD

**Issue by dmlap** [videojs/video.js#4199]
## Description
The loading spinner fails to show up, or disappears after displaying briefly, when playback has temporarily stopped because video data is not available.

## Steps to reproduce
Go to the [advanced example](http://videojs.com/advanced/) on videojs.com.
1. Throttle your connection to "Regular 2G" in Chrome's developer tools.
2. Begin playback.

### Expected
The player stops occasionally because the connection is too slow to support realtime playback. During those periods where playback is stopped, the loading spinner is displayed.

### Actual
The player stops occasionally because the connection is too slow to support realtime playback. The loading spinner either fails to show up entirely or shows up briefly but disappears before playback resumes.

### Error output
No errors.

## Additional Information
Please include any additional information necessary here. Including the following:
#### videojs
5.16.0
#### browsers
Tested in Chrome 56
#### OSes
macOS

**Comment by kassemEzz** [comment]
I faced the same problem and I think I just found out what to do.
I added a custom class on "waiting" event and then removed it on "playing" event.

```javascript
player.on("waiting", function ()
{
    this.addClass("vjs-custom-waiting");
});
player.on("playing", function ()
{
    this.removeClass("vjs-custom-waiting");
});
```
and then added some css values
```css
.vjs-custom-waiting .vjs-loading-spinner
{
    display: block;
}
.video-js.vjs-custom-waiting .vjs-loading-spinner:before,
.video-js.vjs-custom-waiting .vjs-loading-spinner:after
{
    /* I just copied the same animation as in the default css file */
    -webkit-animation: vjs-spinner-spin 1.1s cubic-bezier(0.6, 0.2, 0, 0.8) infinite, vjs-spinner-fade 1.1s linear infinite;
    animation: vjs-spinner-spin 1.1s cubic-bezier(0.6, 0.2, 0, 0.8) infinite, vjs-spinner-fade 1.1s linear infinite; }
}
```

and now the spinner is showing as long as the video is stalled.
I hope that helps.

**Comment by sohrabzia** [comment]
thanks @kassemEzz it worked for me, you need to remove extra closing bracket in css at the end.

**Comment by kassemEzz** [comment]
@sohrabzia glad it worked for u. I removed the extra bracket. Thx

**Comment by saavedrajj** [comment]
@kassemEzz worked for me either. thanks!

**Comment by mysuf** [comment]
I came to report this issue aswell and found this thread. So I can only mention that this bug is also in **v6**. The funny thing is that without knowing about this thread I fixed it almost same as @kassemEzz did (with only vjs-loading class used instead) :) Here is css I used for v6:

```
.vjs-loading:not(.vjs-ad-playing) .vjs-loading-spinner {
    display: block;
    animation: 0s linear .3s forwards a;
}
```
Optionally I added:
```
.vjs-loading:not(.vjs-ad-playing) {
	pointer-events: none;
}
```
To disable any actions until player is ready.

As this is easy-to-fix bug and is opened pretty long time, it would be nice to fix it. Thank you
